backend/bamboo/blueprints/media.py

```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
+import mimetypes
+
+from apiflask import APIBlueprint
+from flask import current_app
+from werkzeug.datastructures import FileStorage
+
+from bamboo.database import db
+from bamboo.database.models import Media
+from bamboo.jobs import gen_small_image
+from bamboo.schemas.media import MediaSchema, IMAGE_SUFFIXES
+from bamboo.utils import gen_uuid
+
+media_bp = APIBlueprint("media", __name__)
+
+
+@media_bp.post("/")
+@media_bp.input(MediaSchema, location="files")
+def upload_media(files_data: dict) -> dict:
+    file: FileStorage = files_data["file"]
+    content_type = mimetypes.guess_type(file.filename)[0]
+    media_dir = Path(current_app.config["BAMBOO_MEDIA_DIR"])
+    media = Media(content_type=content_type, path="")
+    db.session.add(media)
+    # flush to get id
+    db.session.flush()
+    file_suffix = Path(file.filename).suffix.lower()
+    # generate unique filename
+    filename = f"{media.id}_{gen_uuid()[:8]}{file_suffix}"
```

Comment:
I wonder if the id part is necessary. `<uuid>_<filename>` should be sufficient, and it don't need to be committed to DB for the ID.

Reply:
The main purpose is to prevent duplicate names, although it is highly unlikely to occur.